Community/Tdarr_Plugin_Soggys_NVENC_HEVC_CQV_Optimised_Bitrate.js

```diff
@@ -0,0 +1,209 @@
+  response.infoLog += `Optimal Bitrate = ${(optimalVideoBitrate / 1000000).toString()}\n`;
+  response.infoLog += `Stream Bitrate = ${(streamBR / 1000000).toString()}\n`;
+  if (file.ffProbeData.streams[0].codec_name === 'hevc'
+  && (Number.isNaN(optimalVideoBitrate) || Number.isNaN(streamBR) || streamBR < optimalVideoBitrate)) {
+    response.infoLog += '☑File is already in hevc and is below optimal bitrate!\n';
+    return response;
+  }
+  if (file.ffProbeData.streams[0].codec_name === 'hevc'
+  && (file.video_resolution === '480p' || file.video_resolution === '576p' || file.video_resolution === '720p')) {
+    response.infoLog += '☑File is low res, wont transcode!\n';
+    return response;
+  }
+
+  // Check if preset is configured, default to slow if not
+  let ffmpeg_preset;
+  if (!inputs.ffmpeg_preset) {
+    ffmpeg_preset = 'slow';
+    response.infoLog += '☑Preset not set, defaulting to slow\n';
+  } else {
+    ffmpeg_preset = inputs.ffmpeg_preset;
+    response.infoLog += `☑Preset set as ${inputs.ffmpeg_preset}\n`;
+  }
+
+  // Set Subtitle Var before adding encode cli
+  for (let i = 0; i < file.ffProbeData.streams.length; i += 1) {
+    try {
+      if (
+        file.ffProbeData.streams[i].codec_name.toLowerCase() === 'mov_text'
+        && file.ffProbeData.streams[i].codec_type.toLowerCase() === 'subtitle'
+      ) {
+        subcli = '-c:s srt';
+      }
+    } catch (err) {
+      response.infoLog += 'Error reading subtitle codec!';
+    }
+    // mitigate TrueHD audio causing Too many packets error
+    try {
+      if (
+        file.ffProbeData.streams[i].codec_name.toLowerCase() === 'truehd'
+        || (file.ffProbeData.streams[i].codec_name.toLowerCase() === 'dts'
+          && file.ffProbeData.streams[i].profile.toLowerCase() === 'dts-hd ma')
+        || (file.ffProbeData.streams[i].codec_name.toLowerCase() === 'aac'
+          && file.ffProbeData.streams[i].sample_rate.toLowerCase() === '44100'
+          && file.ffProbeData.streams[i].codec_type.toLowerCase() === 'audio')
+      ) {
+        maxmux = ' -max_muxing_queue_size 9999';
+      }
+    } catch (err) {
+      response.infoLog += 'Error reading audio codec!';
+    }
+    // mitigate errors due to embeded pictures
+    try {
+      if (
+        (file.ffProbeData.streams[i].codec_name.toLowerCase() === 'png'
+          || file.ffProbeData.streams[i].codec_name.toLowerCase() === 'bmp'
+          || file.ffProbeData.streams[i].codec_name.toLowerCase() === 'mjpeg')
+        && file.ffProbeData.streams[i].codec_type.toLowerCase() === 'video'
+      ) {
+        map = '-map 0:v:0 -map 0:a -map 0:s?';
+      }
+    } catch (err) {
+      response.infoLog += 'Error reading image codec!';
+    }
+  }
+
+  response.preset += `-hwaccel cuda ,${map} -dn -c:v hevc_nvenc -b:v 0 -preset ${ffmpeg_preset} -cq ${inputs.cqv}`
+    + ` -rc-lookahead 32 -bf ${inputs.bframe} -a53cc 0 -c:a copy ${subcli}${maxmux}`;
+
+  if (inputs.ten_bit === 'true') {
```

Comment:
Changing the type to boolean, this can be simplified to `if (inputs.ten_bit)` :)